E0109 needs to be updated to new format

From: src/test/compile-fail/E0109.rs

Error E0109 needs a span_label, updating it from:

```
error[E0109]: type parameters are not allowed on this type
  --> src/test/compile-fail/E0109.rs:11:14
   |
11 | type X = u32<i32>; //~ ERROR E0109
   |              ^^^
```

To:

```
error[E0109]: type parameters are not allowed on this type
  --> src/test/compile-fail/E0109.rs:11:14
   |
11 | type X = u32<i32>; //~ ERROR E0109
   |              ^^^ type parameter not allowed
```

--- a/src/librustc/middle/astconv_util.rs
+++ b/src/librustc/middle/astconv_util.rs
@@ -24,8 +24,10 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
     pub fn prohibit_type_params(self, segments: &[ast::PathSegment]) {
         for segment in segments {
             for typ in segment.parameters.types() {
-                span_err!(self.sess, typ.span, E0109,
-                          "type parameters are not allowed on this type");
+                struct_span_err!(self.sess, typ.span, E0109,
+                                 "type parameters are not allowed on this type")
+                    .span_label(typ.span, &format!("type parameter not allowed"))
+                    .emit();
                 break;
             }
             for lifetime in segment.parameters.lifetimes() {

--- a/src/test/compile-fail/E0109.rs
+++ b/src/test/compile-fail/E0109.rs
@@ -9,6 +9,7 @@
 // except according to those terms.
 
 type X = u32<i32>; //~ ERROR E0109
+                   //~| NOTE type parameter not allowed
 
 fn main() {
 }

